app/services/importers/fe_csv.rb

```diff
@@ -0,0 +1,79 @@
+require 'csv'
+
+module Importers
+  class FeCsv
+    attr_reader :path
+
+    def initialize(path_to_csv:)
+      @path = path_to_csv
+    end
+
+    # UKPRN
+    # URN (if applicable)
+    # Provider name
+    # Provider type
+    # Contact name
+    # Contact email
+    # Contact phone
+    # Contact address (full)
+    # Main address line 1
+    # Main address line 2
+    # Main address line 3
+    # Main town / city
+    # Main county
+    # Main postcode
+
+    def call
+      rows.each do |row|
+        school = School.find_by(ukprn: row['UKPRN'])
+
+        if school.nil?
+          rb = FurtherEducationCollege.create!(
+            name: row['Provider name'],
+            organisation_type: 'FurtherEducationSchool',
+            who_will_order_devices: 'responsible_body',
+            address_1: row['Main address line 1'],
+            address_2: row['Main address line 2'],
+            address_3: row['Main address line 3'],
+            town: row['Main town / city'],
+            county: row['Main county'],
+            postcode: row['Main postcode'],
+          )
+
+          school = FurtherEducationSchool.create!(
+            responsible_body: rb,
+            ukprn: row['UKPRN'],
+            name: row['Provider name'],
+            address_1: row['Main address line 1'],
+            address_2: row['Main address line 2'],
+            address_3: row['Main address line 3'],
+            town: row['Main town / city'],
+            county: row['Main county'],
+            postcode: row['Main postcode'],
+          )
+        end
+
+        contact = school.contacts.first_or_create!(
+          email_address: row['Contact email'],
+          full_name: row['Contact name'],
+          role: 'contact',
+          title: nil,
+          phone_number: row['Contact phone'],
+        )
+
+        next if school.preorder_information.present?
+
+        school.create_preorder_information!(
```

Comment:
Do we need to prevent FE colleges from being able to modify this in the UI?

Reply:
None of the frontend changes have been done yet. But yes if we want to keep everything on the RB we can remove it from the interface. If we want to do this we may also want to remove some other features such as being able to add users on the school.

Maybe `/schools/123456` will link to the current controller and `/schools/12345678` can link of to another controller and set of views